README.md

```diff
@@ -46,6 +46,26 @@ watcher.on('delete', function (filepath, root) { console.log('file deleted', fil
 watcher.close();
 ```
 
+In addition to change events, the `watchman`-specific watcher may emit a
+`fresh_instance` event that signals the state got lost and we will not receive
+notifications for some of the changes. When this happens it is recommended to
+empty or reset any cache that tracks the state of the files. The event is
+immediately followed by change events for each file that does exist, that
+you can leverage to fill a cache again.
+
+For example, say we want to write an application that keeps track of file paths
+in a `Set`. Here is a possible way to handle it:
+
+```js
+function findFiles() { return new Set(glob('**/*.js')); }
+var files = findFiles();
+var watcher = sane('.', {glob: '**/*.js'});
+
+watcher.on('fresh_instance', () => { files = findFiles(); });
```

Comment:
and this can invalidate the cache by doing something like `files.clear()` or `files = new Set()`.  The `add` event below will trigger to fill it up and save you from performing a `glob` call to walk the filesystem

Reply:
Sounds good for that one case.